apps/web/middleware.ts

```diff
@@ -11,6 +11,9 @@ const V2_WHITELIST = [
   "/availability",
   "/bookings",
   "/event-types",
+  // Apps contains trailing slash to prevent app overview from being rendered as v2,
+  // since it doesn't exist yet.
+  "/apps/",
```

Comment:
This actually works fine for app pages but it gives an error when navigating to app categories, such as http://localhost:3000/apps/categories/video. We may want to change the logic and rely on regex, WDYT @zomars?

Reply:
Something like:

```javascript
const V2_WHITELIST = [
  "/settings/admin",
  "/settings/my-account/[a-z](-?[a-z])*",
  "/availability",
  "/bookings/[a-z](-?[a-z])*",
  "/event-types",
  "/apps/[a-z](-?[a-z])*",
];

// ...

if (
    req.cookies.has("calcom-v2-early-access") &&
    V2_WHITELIST.some((p) => new RegExp(`^${p.replace("/", "\\/")}$`).test(url.pathname))
  ) {
    // rewrite to the current subdomain under the pages/sites folder
    url.pathname = `/v2${url.pathname}`;
    return NextResponse.rewrite(url);
  }
```

Reply:
Oh my, I just noticed "Routing Forms" app has the wrong slug, it uses underscore "routing_forms", we will need to standardize that or adjust the regex 😅

Reply:
I like the regex solution as well. Think this is a lot safer!

Reply:
This logic will be gone post v2 launch so I'm okay on adding individual routes and not over engineer this logic.